File: CheckIn.Api.Common/Models/Lists/TaskListModel.cs
using CheckIn.Api.Common.Enums;
using CheckIn.Api.Common.Models.Interfaces;


namespace CheckIn.Api.Common.Models.Lists;

public record TaskListModel : IEntityModel
{
    public Guid Id { get; init; }
    public required string Title { get; init; }
    public string? Notes { get; init; }

    public required string Hash { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime LastModifiedAt { get; init; }
    public DateTime DeadLine { get; init; }
    public TaskState State { get; init; }
    public Guid CreatedById { get; init; }
    public SubtaskMode SubtaskMode { get; init; }
    public bool RequiresAuthenticationToComplete { get; init; } = true;
}

File: checkin_frontend/lib/core/models/task/task_list_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../enums/task_enums.dart';

part 'task_list_model.freezed.dart';
part 'task_list_model.g.dart';

@freezed
sealed class TaskListModel with _$TaskListModel {
  const factory TaskListModel({
    required String id,
    required String title,
    String? notes,
    required String hash,
    required DateTime createdAt,
    required DateTime deadLine,
    required String createdById,
    required TaskState state,
    required SubtaskMode subtaskMode,
    @Default(true) bool requiresAuthenticationToComplete,
    String? allowedDomain,
  }) = _TaskListModel;

  factory TaskListModel.fromJson(Map<String, dynamic> json) => _$TaskListModelFromJson(json);
}